pandas/io/pytables.py

```diff
@@ -313,10 +313,14 @@ def read_hdf(path_or_buf, key=None, **kwargs):
 
         """
 
-    if kwargs.get('mode', 'a') not in ['r', 'r+', 'a']:
+    # Ignore any supplied mode, we only need to read data
```

Comment:
This is actually tricky, because IIRC HDF5 has some locking issues when some files are read-only and others are append, which is why the default is append (in a multi-open e.g. process/thread context).